server/plugin/api.go

```diff
@@ -220,6 +229,43 @@ func (p *Plugin) handleUnlinkProject(w http.ResponseWriter, r *http.Request) {
 	p.writeJSON(w, &successResponse)
 }
 
+func (p *Plugin) handleDeleteAllSubscriptions(mattermostUserID, projectID string) (int, error) {
+	subscriptionList, err := p.Store.GetAllSubscriptions(mattermostUserID)
+	if err != nil {
+		p.API.LogError(constants.FetchSubscriptionListError, "Error", err.Error())
+		return http.StatusInternalServerError, err
+	}
+
+	for _, subscription := range subscriptionList {
+		if subscription.ProjectID == projectID && subscription.MattermostUserID == mattermostUserID {
```

Comment:
if we are fetching all subscriptions of a user with the id `mattermostUserID`. Do we need to check the id here again?